src/main/java/com/deloitte/elrr/datasync/scheduler/LRSSyncSchedulingService.java

```diff
@@ -56,22 +54,33 @@ public void run() {
 
             // If no import record
             if (importRecord == null) {
-                importRecord = createImport();
+                importRecord = importService.createImport();
+            } else if (importRecord.getRecordStatus().equalsIgnoreCase(
+                    StatusConstants.INPROCESS)) {
+                log.info("Statements are still being processed.");
+                return;
             }
 
             Statement[] result = null;
-            importRecord = updateImportInProcess(importRecord);
+
+            // Update import start and end dates
+            importRecord = importService.updateImportStartEndDates(
+                    importRecord);
+
+            // Update import status to INPROCESS
+            importRecord = importService.updateImportStatus(importRecord,
+                    StatusConstants.INPROCESS);
```

Comment:
ohh i see its a string not an enum. perhaps convert to enum with a string output.